libdrgn/python/thread.c

```diff
@@ -17,6 +17,22 @@ static PyObject *Thread_get_prstatus(Thread *self)
 					 self->thread->prstatus.len);
 }
 
+static PyObject *Thread_get_object(Thread *self)
+{
+	if (!((self->prog->prog.flags & DRGN_PROGRAM_IS_LINUX_KERNEL) &&
+	      (self->prog->prog.flags & DRGN_PROGRAM_IS_LIVE))) {
+		PyErr_SetString(
+			PyExc_ValueError,
```

Comment:
I'm wondering if `AttributeError` would be more appropriate. That's what we do for `drgn.Type` fields that don't exist for a particular type.

Reply:
This would be inconsistent with your feedback here: https://github.com/osandov/drgn/pull/129#discussion_r755609382

Would you like for me to change both to `AttributeError`, or keep them both as `ValueError`?

Reply:
The link isn't working for me, which comment was that? `ValueError` is probably fine here too, though.

Reply:
From 20 days ago, under _drgn.pyi: with the message

> This should probably raise an exception if the target isn't a core dump. Probably ValueError (which is what we translate DRGN_ERROR_INVALID_ARGUMENT to). #

Reply:
These cases are a bit different. `Program.crashed_thread()` is a method which always exists, but only makes sense for some inputs. `Thread.object` is an attribute which only makes sense for some threads, so one could argue that it only exists in some cases. If this were implemented in pure Python, it might look something like:

```python
class Program:
    def crashed_thread(self):
        if self is not a core dump:
            raise ValueError("not a core dump")

def create_thread(prog, ...):
    thread = Thread(...)
    if prog is the Linux kernel:
        prog.object = task
```

Maybe `Thread.object` should be a method rather than an attribute, though. I'm not sure.

Reply:
Ah, I see. Makes sense, I'll change this to a `AttributeError`, thanks for the explanation.